platform/server/elasticsearch/ElasticsearchService.ts

```diff
@@ -74,7 +80,28 @@ export class ElasticsearchService implements CoreService {
     }
   }
 
-  getClusterOfType$(type: ElasticsearchClusterType) {
-    return k$(this.clusters$)(map(clusters => clusters[type]));
+  getAdminClient$() {
+    return k$(this.clients$)(map(clients => new AdminClient(clients.admin)));
   }
+
+  getScopedDataClient$(headers: Headers) {
+    return k$(this.clients$)(map(clients => new ScopedDataClient(clients.data, headers)));
+  }
+
+  getScopedDataClient(headers: Headers) {
+    return k$(this.getScopedDataClient$(headers))(first(), toPromise());
+  }
+
+  // typed as a DataClient
+  // for health check, e.g. which uses callWithInternalUser
+  // const unscopedDataClient = elasticsearch.getUnscopedDataClient();
+
+  // SAML or Basic Auth case
+  // scoped adminClient
+  // const scopedAdminClient = elasticsearch.getScopedAdminClient(headers);
```

Comment:
__note:__ not relevant to this PR, but for `auth` use cases it will look more like:
`getScopedClient('security', headers)`.

Reply:
++ we need to look into clients for `security` and `monitoring`. We might want a `createClient` type helper. I don't think it should be registered in the `ElasticsearchService` and have a `get`-er there. The `security` client is only used within the Security plugin, so it should create its own client that it manages itself (but we likely want to expose helpers to create those clients).